src/DynamoPackages/PackageLoader.cs

```diff
@@ -278,11 +279,27 @@ public void LoadAll(LoadPackageParams loadPackageParams)
         public void LoadPackages(IEnumerable<Package> packages)
         {
             var enumerable = packages.ToList();
+
+            // This fix is in reference to the crash reported in task: https://jira.autodesk.com/browse/DYN-2101
+            // TODO: we will be re-evaluating this workflow, to find the best clean solution. 
+
+            // The reason for this crash is, when a new package is being loaded into the dynamo, it will reload 
+            // all the libraries into the VM. Since the graph execution runs are triggered asynchronously, it causes 
+            // an exception as the VM is reinitialized during the execution run. To avoid this, we disable the execution run's that
+            // are triggered while the package is still being loaded. Once the package is completely loaded and the VM is reinitialized,
+            // a final run is triggered that would execute the nodes in the workspace after resolving them.  
+
+            // Disabling the run here since new packages are being loaded. 
+            EngineController.DisableRun = true;
+
             foreach (var pkg in enumerable)
             {
                 TryLoadPackageIntoLibrary(pkg);
             }
 
+            // Setting back the DisableRun property back to false, as the package loading is completed.
+            EngineController.DisableRun = false;
```

Comment:
is there ever a case where we could accidentally turn runs off and then never turn them back on?

Reply:
I don't think so as this would be set back to false, once the package loading is completed.

Reply:
I think @mjkkirschner was alluding to a case where an exception is thrown from `TryLoadPackageIntoLibrary` and this flag-reset step is skipped in that case. I think `TryLoadPackageIntoLibrary` has a `try-catch` inside it that can catch any exception and not rethrow it in order for this flag reset to take place right after but to be safe we can have another `try-catch-finally` around the `for` loop and reset the flag in the `finally` block to ensure it always happens no matter what.

Reply:
@aparajit-pratap thats exactly what I was thinking - do you think it's worth doing this:
or should it be okay since TryLoadPackageIntoLibrary will already catch and move onto the next package.

```suggestion

try {
      foreach (var pkg in enumerable)
            {
                TryLoadPackageIntoLibrary(pkg);
            }
      }
finally{
            EngineController.DisableRun = false;
           }
```

Reply:
It would at least be worthwhile checking that the exception caught inside `TryLoadPackageIntoLibrary` is not re-thrown from the `Log` methods?

Reply:
log doesn't seem to rethrow anything.